### PR TITLE
implement scr_have_restart binary

### PR DESCRIPTION
**Pull request by mmpozulp**
TODO:
1. integrate scr_have_restart binary build with CMake
2. add error checking to scr_have_restart binary

### CODE REVIEW

**Comment by mmpozulp** [comment]
@gonsie @becker33 How do I add you two as reviewers? Thanks for helping me develop my GH-fu :)

**Comment by becker33** [comment]
@mmpozulp Tagging us is probably plenty, I don't know whether you can user the reviewers tab on the right hand side without being a member of the project on github.

Please make sure you've checked the box that allows us to make changes to your branch, as we will need to add to this PR before it is ready to merge.

**Comment by mmpozulp** [comment]
"Allow edits from maintainers" is checked.

**Comment by becker33** [comment]
@gonsie it didn't use sed in place, so no issues there. I've confirmed the behavior is the same on linux and OS X.

**Comment by adammoody** [comment]
@becker33 Can you try a test where an app may use a quoted argument?

./app 'foo bar'

Some applications do this kind of thing.  The app should see this as a single argument string "foo bar" in its argv list.

It may be that we don't handle this well already.  I haven't checked that case before.

**Comment by becker33** [comment]
I believe what we currently have will handle that properly, and that in the new case it will only require that they properly nest the quotes. I will make sure to test this.

**Comment by mmpozulp** [comment]
It works! Thanks :)

**Comment by adammoody** [comment]
@becker33, let's replicate the spaces commit for the other RM types

**Comment by becker33** [comment]
@adammoody yes that's on today's agenda.

File: src/scr_have_restart.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    rc = MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    if (rc != MPI_SUCCESS) {
+        fprintf(stderr, "%s: MPI_Comm_rank failed %d", myname, rc);
+    }
+
+    if ( (rank == 0) && (flag > 0) ) {
+        printf("%s\n", fname);
+    }
+}
```
Comment (gonsie) [inline src/scr_have_restart.c]: Is there any cleanup that needs to happen? SCR Finalize or MPI Finalize?

**Pull request merged**